packages/core/src/analytics/dispatch-emit.ts

```diff
@@ -8,13 +7,20 @@ export const dispatchAndEmit = async (
   try {
     const ctx = await dispatch(event, queue, emitter, options)
     if (ctx.failedDelivery()) {
-      throw ctx
+      emitter.emit('error', {
+        code: 'delivery_failure',
+        message: 'failed to deliver event',
+        ctx: ctx,
+      })
     } else {
       emitter.emit(event.type, ctx)
       return ctx
     }
   } catch (err) {
-    emitter.emit('error', err as CoreContext)
-    return err
```

Comment:
`dispatchAndEmit` is not used outside Node ... right? Just wondering if removing the `return err` would break anything. If not, all good 👍

Reply:
Yes, this is a node only method (for now) -- but I decided on putting it in core because I thought it might make sense to have browser adopt it.